### PR TITLE
feat: Heatmaps UI alpha

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

Draft approach at https://github.com/PostHog/posthog/issues/21531

This is a very WIP thing. Wanted to merge what I have so far so we can at least test the concept in production of running an embedded iframe before I go any further with it

## Changes
Screenshot 2024-04-15 at 12 04 34

- [x] Adds a flagged beta UI for heatmaps
- [x] Allows searching existing URLs to load
- [x] Only allow loading authorized URLs (and give a way to change that easily here)

- [ ] Add controls to the main UI so that we can sync over iframe messages
- [ ] Detect if the toolbar and show a message if not and how to troubleshoot (sending them to the toolbar instead)

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

File: frontend/src/lib/components/HedgehogBuddy/HedgehogBuddy.tsx [hunk 1]
```diff
@@ -145,8 +152,8 @@ export class HedgehogActor {
             const duration = this.animationIterations
                 ? this.animationIterations * this.animation.frames * (1000 / FPS)
                 : '∞'
-            // eslint-disable-next-line no-console
-            console.log(`Hedgehog: Will '${this.animationName}' for ${duration}ms`)
+
+            this.log(`Will '${this.animationName}' for ${duration}ms`)
```
Comment (benjackwhite) [inline frontend/src/lib/components/HedgehogBuddy/HedgehogBuddy.tsx:156]: This finally got on my nerves enough to remove

File: frontend/src/scenes/heatmaps/HeatmapsBrowser.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+                <p className="p-2">Controls</p>
+            </div> */}
+
+            <div className="flex flex-col overflow-hidden w-full h-[90vh] rounded border">
+                <div className="bg-accent-3000 p-2 border-b flex items-center gap-2">
+                    <span className="flex-1">
+                        <LemonInputSelect
+                            mode="single"
+                            allowCustomValues
+                            placeholder="e.g. https://your-website.com/pricing"
```
Comment (pauldambra) [inline frontend/src/scenes/heatmaps/HeatmapsBrowser.tsx:31]: you could spark joy by picking a random URL for their team from the sessions table?

File: frontend/src/scenes/heatmaps/heatmapsBrowserLogic.ts [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+            },
+        ],
+
+        topUrls: [
+            null as { url: string; count: number }[] | null,
+            {
+                loadTopUrls: async () => {
+                    const query: HogQLQuery = {
+                        kind: NodeKind.HogQLQuery,
+                        query: hogql`SELECT properties.$current_url AS url, count(1) as count FROM events
```
Comment (pauldambra) [inline frontend/src/scenes/heatmaps/heatmapsBrowserLogic.ts:72]:
```suggestion
                        query: hogql`SELECT properties.$current_url AS url, count() as count FROM events
```

i don't know if they ignore the `(1)` but clickhouse docs say `count()` has special optimizations

https://clickhouse.com/docs/en/sql-reference/aggregate-functions/reference/count

**Review by pauldambra — APPROVED** [review]
that's really cool

File: frontend/src/scenes/heatmaps/HeatmapsBrowser.tsx (hunk 2 above)
Comment (pauldambra) [reply frontend/src/scenes/heatmaps/HeatmapsBrowser.tsx:31]: was reviewing it and then realised.. we just need to smash it in and then pick away at it

**Pull request merged**